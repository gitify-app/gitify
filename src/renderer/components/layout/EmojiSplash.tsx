import { Box, Stack } from '@primer/react';
import type { FC } from 'react';
import { EmojiText } from '../primitives/EmojiText';
import { Centered } from './Centered';

interface IEmojiSplash {
  emoji: string;
  heading: string;
  subHeadings?: string[];
  fullHeight?: boolean;
}

export const EmojiSplash: FC<IEmojiSplash> = ({
  fullHeight = true,
  subHeadings = [],
  ...props
}: IEmojiSplash) => {
  return (
    <Centered fullHeight={fullHeight}>
      <Stack
        align="center"
        direction="vertical"
        gap="condensed"
        justify={'center'}
      >
        <Stack align="center" direction="vertical" gap="spacious">
          <EmojiText text={props.emoji} />
          <Box className="text-xl font-semibold">{props.heading}</Box>
        </Stack>

        {subHeadings.map((description, i) => {
          return (
            // biome-ignore lint/suspicious/noArrayIndexKey: using index for key to keep the error constants clean
            <Box key={`error_description_${i}`} className="text-center">
              {description}
            </Box>
          );
        })}
      </Stack>
    </Centered>
  );
};
