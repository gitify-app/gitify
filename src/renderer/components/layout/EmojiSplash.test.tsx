import { render } from '@testing-library/react';
import { EmojiSplash } from './EmojiSplash';

describe('renderer/components/layout/EmojiSplash.tsx', () => {
  it('should render itself & its children - heading only', () => {
    const tree = render(<EmojiSplash emoji="🍺" heading="Test Heading" />);

    expect(tree).toMatchSnapshot();
  });

  it('should render itself & its children - heading and sub-heading', () => {
    const tree = render(
      <EmojiSplash
        emoji="🍺"
        heading="Test Heading"
        subHeadings={['Test Sub-Heading']}
      />,
    );

    expect(tree).toMatchSnapshot();
  });
});
