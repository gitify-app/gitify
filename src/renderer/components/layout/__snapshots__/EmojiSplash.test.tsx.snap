// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer/components/layout/EmojiSplash.tsx should render itself & its children - heading and sub-heading 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Stack__StyledStack-sc-x3xa2i-0 duhwY h-full"
        data-align="center"
        data-direction="vertical"
        data-justify="center"
        data-padding="spacious"
        data-wrap="nowrap"
      >
        <div
          class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
          data-align="center"
          data-direction="vertical"
          data-gap="condensed"
          data-justify="center"
          data-padding="none"
          data-wrap="nowrap"
        >
          <div
            class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
            data-align="center"
            data-direction="vertical"
            data-gap="spacious"
            data-justify="start"
            data-padding="none"
            data-wrap="nowrap"
          >
            <div
              class="Box-sc-g0xbh4-0 text-5xl"
            >
              <img
                alt="🍺"
                class="emoji"
                draggable="false"
                src="images/twemoji/1f37a.svg"
              />
            </div>
            <div
              class="Box-sc-g0xbh4-0 text-xl font-semibold"
            >
              Test Heading
            </div>
          </div>
          <div
            class="Box-sc-g0xbh4-0 text-center"
          >
            Test Sub-Heading
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Stack__StyledStack-sc-x3xa2i-0 duhwY h-full"
      data-align="center"
      data-direction="vertical"
      data-justify="center"
      data-padding="spacious"
      data-wrap="nowrap"
    >
      <div
        class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
        data-align="center"
        data-direction="vertical"
        data-gap="condensed"
        data-justify="center"
        data-padding="none"
        data-wrap="nowrap"
      >
        <div
          class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
          data-align="center"
          data-direction="vertical"
          data-gap="spacious"
          data-justify="start"
          data-padding="none"
          data-wrap="nowrap"
        >
          <div
            class="Box-sc-g0xbh4-0 text-5xl"
          >
            <img
              alt="🍺"
              class="emoji"
              draggable="false"
              src="images/twemoji/1f37a.svg"
            />
          </div>
          <div
            class="Box-sc-g0xbh4-0 text-xl font-semibold"
          >
            Test Heading
          </div>
        </div>
        <div
          class="Box-sc-g0xbh4-0 text-center"
        >
          Test Sub-Heading
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renderer/components/layout/EmojiSplash.tsx should render itself & its children - heading only 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Stack__StyledStack-sc-x3xa2i-0 duhwY h-full"
        data-align="center"
        data-direction="vertical"
        data-justify="center"
        data-padding="spacious"
        data-wrap="nowrap"
      >
        <div
          class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
          data-align="center"
          data-direction="vertical"
          data-gap="condensed"
          data-justify="center"
          data-padding="none"
          data-wrap="nowrap"
        >
          <div
            class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
            data-align="center"
            data-direction="vertical"
            data-gap="spacious"
            data-justify="start"
            data-padding="none"
            data-wrap="nowrap"
          >
            <div
              class="Box-sc-g0xbh4-0 text-5xl"
            >
              <img
                alt="🍺"
                class="emoji"
                draggable="false"
                src="images/twemoji/1f37a.svg"
              />
            </div>
            <div
              class="Box-sc-g0xbh4-0 text-xl font-semibold"
            >
              Test Heading
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Stack__StyledStack-sc-x3xa2i-0 duhwY h-full"
      data-align="center"
      data-direction="vertical"
      data-justify="center"
      data-padding="spacious"
      data-wrap="nowrap"
    >
      <div
        class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
        data-align="center"
        data-direction="vertical"
        data-gap="condensed"
        data-justify="center"
        data-padding="none"
        data-wrap="nowrap"
      >
        <div
          class="Stack__StyledStack-sc-x3xa2i-0 duhwY"
          data-align="center"
          data-direction="vertical"
          data-gap="spacious"
          data-justify="start"
          data-padding="none"
          data-wrap="nowrap"
        >
          <div
            class="Box-sc-g0xbh4-0 text-5xl"
          >
            <img
              alt="🍺"
              class="emoji"
              draggable="false"
              src="images/twemoji/1f37a.svg"
            />
          </div>
          <div
            class="Box-sc-g0xbh4-0 text-xl font-semibold"
          >
            Test Heading
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
